src/tm-threads.h

```diff
@@ -253,6 +253,10 @@ static inline void TmThreadsCaptureHandleTimeout(ThreadVars *tv, Packet *p)
         tv->tmqh_out(tv, p);
 }
 
+/**
+ * Note: Only ever called if tv->break_loop is true and
```

Comment:
is this is correct all the if's below could be (debug) asserts

Reply:
For my local fix, I simply removed the ` if (unlikely(!tv->break_loop))` in `TmThreadsCaptureBreakLoop` and changed the `} else if (detect_tvs[i]->break_loop) {` to `} else {`.

I think that is better as:
1. the logic of break loop is still within a single function.
2. there are already extra checks of `tv->break_loop` in other use cases.
3. the `if (tm->PktAcqLoop && tm->PktAcqBreakLoop)` basically ensures `tv->break_loop` is true.

Reply:
I agree it is better to keep the capture specific logic out of the detect engine as much as possible.